src/timesync/timesyncd-bus.c

```diff
@@ -55,6 +56,46 @@ static int property_get_servers(
         return sd_bus_message_close_container(reply);
 }
 
+static int property_get_current_server_name(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        ServerName **s = userdata;
+
+        assert(s);
+        assert(bus);
+        assert(reply);
+
+        return sd_bus_message_append(reply, "s", *s ? (*s)->string : "");
+}
+
+static int property_get_current_server_address(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        _cleanup_free_ char *pretty = NULL;
+        ServerAddress **a = userdata;
+
+        assert(a);
+        assert(bus);
+        assert(reply);
+
+        if (*a)
+                (void) server_address_pretty(*a, &pretty);
+
+        return sd_bus_message_append(reply, "s", strempty(pretty));
```

Comment:
so far we exposed IP addresses generally in binary, see resolved for example, we should do that too here, i think

Reply:
OK. Done.